pkg/util/util.go

```diff
@@ -302,3 +305,24 @@ func WaitUntilTimeout(name string, fn func(), timeout time.Duration) {
 	case <-exited:
 	}
 }
+
+// WriteFileAtomic writes file to temp and atomically move when everything else succeeds.
+func WriteFileAtomic(filename string, data []byte, perm os.FileMode) error {
+	dir, name := path.Dir(filename), path.Base(filename)
+	f, err := os.CreateTemp(dir, name)
+	if err != nil {
+		return err
+	}
+	n, err := f.Write(data)
+	f.Close()
+	if err == nil && n < len(data) {
+		err = io.ErrShortWrite
+	} else {
```

Comment:
err != nil  will run into this branch and we lost the origin err

Reply:
I copy this code snippet from 

https://github.com/siddontang/go/blob/bdc77568d726a8702315ec4eafda030b6abc4f43/ioutil2/ioutil.go#L15-L33

.. but it's fine we fix it on picking.